addon/validations/factory.js

```diff
@@ -102,11 +104,15 @@ function setOwner(obj, model) {
 export default function buildValidations(validations = {}) {
   processDefaultOptions(validations);
 
-  const Validations = createValidationsObject(validations);
+  var Validations;
 
   return Ember.Mixin.create({
     validations: computed(function() {
-      return Validations.create({ model: this, _parentValidations: this._super() });
+      if(!Validations) {
+        // We only want to create this once per class
+        Validations = createValidationsClass(this._super(), validations);
```

Comment:
I believe we may want to split this into 2 CP, `validationClass` and `validation`, as we may risk a memory leak by preserving the result of (or  member of) `_super()` which is an instance of the supers Validation class, rather then the class. The instance is allow to store instance specific state, and Im slightly worried that the `createValidationsClass` method may accidentally pass along some of that instance state.

Maybe something like:
```js
let Validations;
...
validationsClass: computed(function() {
  if (!Validations) {
     Validations = createVallidationClass(this._super(), validations)
  },
  
  return Validations
}),

validations: computed(function() {
   return this.get('validationClass').create({ model: this })
})
```

Thoughts?